src/App.js

```diff
@@ -13,111 +14,100 @@ import NotificationsPortal from '@redhat-cloud-services/frontend-components-noti
 
 export const PermissionContext = createContext();
 
-class App extends Component {
-  unregister;
-  constructor() {
-    super();
-    this.state = {
-      readPermission: undefined,
-      writePermission: undefined,
-      executePermission: undefined,
-      isReceptorConfigured: undefined,
-      arePermissionLoaded: false,
-      hasSmartManagement: undefined,
-    };
-  }
+const App = (props) => {
+  const chrome = useChrome();
+  const [
+    { readPermission, writePermission, executePermission, arePermissionLoaded },
+    setPermissions,
+  ] = useState({
+    readPermission: undefined,
+    writePermission: undefined,
+    executePermission: undefined,
+    arePermissionLoaded: false,
+  });
+  const [isReceptorConfigured, setIsReceptorConfigured] = useState(undefined);
+  const [hasSmartManagement, setHasSmartManagement] = useState(undefined);
 
-  handlePermissionUpdate = (hasRead, hasWrite, hasExecute) =>
-    this.setState({
+  const handlePermissionUpdate = (hasRead, hasWrite, hasExecute) =>
+    setPermissions({
       readPermission: hasRead,
       writePermission: hasWrite,
       executePermission: hasExecute,
       arePermissionLoaded: true,
     });
 
-  componentWillUnmount() {
-    if (typeof this.unregister === 'function') {
-      this.unregister();
-    }
-  }
-  async componentDidMount() {
-    insights.chrome.init();
-    insights.chrome?.hideGlobalFilter?.();
-    insights.chrome.identifyApp('remediations');
-    // wait for auth first, otherwise the call to RBAC may 401
-    await window.insights.chrome.auth.getUser().then((user) =>
-      this.setState({
-        hasSmartManagement: user.entitlements.smart_management.is_entitled,
-      })
-    );
-    getIsReceptorConfigured().then((isConfigured) =>
-      this.setState({
-        isReceptorConfigured: isConfigured.data.length > 0,
-      })
-    );
-    this.unregister = insights.chrome.on('APP_NAVIGATION', (event) => {
-      if (typeof event?.domEvent?.href === 'string') {
-        this.props.history.push('/');
-      }
-    });
-    window.insights.chrome
-      .getUserPermissions('remediations')
-      .then((remediationsPermissions) => {
-        const permissionList = remediationsPermissions.map(
-          (permissions) => permissions.permission
+  useEffect(() => {
+    let unregister;
+    if (chrome) {
+      chrome.identifyApp('remediations');
+      chrome?.hideGlobalFilter();
```

Comment:
optional chaining must be present be up to the function call. otherwise it will raise error for undefined
```suggestion
      chrome?.hideGlobalFilter?.();
```